projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -25,6 +25,30 @@ public function __construct( $feed ) {
 		$this->feed = esc_url_raw( $feed );
 	}
 
+	/**
+	 * Retrieves track quantity.
+	 *
+	 * @returns Integer number of tracks
+	 */
+	public function get_track_quantity() {
+		$rss = $this->load_feed();
+
+		if ( is_wp_error( $rss ) ) {
+			return $rss;
+		}
+
+		/**
+		 * Allow requesting a specific number of tracks from SimplePie's `get_items` call.
+		 * The default number of tracks is ten.
+		 *
+		 * @since 9.5.0
+		 *
+		 * @param int    $number Number of tracks fetched. Default is 10.
+		 * @param object $rss    The SimplePie object built from core's `fetch_feed` call.
+		 */
+		return apply_filters( 'jetpack_podcast_helper_list_quantity', 10, $rss );
```

Comment:
@jeherve The previous filter usage supplied the `$rss` SimplePie object, which makes it awkward when this filter is extracted out to a separate method.

What would you think about omitting the `$rss` arg from the `jetpack_podcast_helper_list_quantity` filter to simplify things? And _if_ it is removed, is there a certain annotation/practice for modifying filters like that? I can't imagine many would be using the `$rss` arg but 🤷

Reply:
It's indeed not ideal. There are different ways we could go about this, but I think the safest would be to start deprecating `jetpack_podcast_helper_list_quantity`, and introduce a new filter without `$rss`. 

Something like this:

```diff
diff --git a/projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php b/projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php
index b23620d7ba..c79d047ce0 100644
--- a/projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php
+++ b/projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php
@@ -28,25 +28,18 @@ class Jetpack_Podcast_Helper {
 	/**
 	 * Retrieves track quantity.
 	 *
-	 * @returns Integer number of tracks
+	 * @returns int number of tracks
 	 */
 	public function get_track_quantity() {
-		$rss = $this->load_feed();
-
-		if ( is_wp_error( $rss ) ) {
-			return $rss;
-		}
-
 		/**
 		 * Allow requesting a specific number of tracks from SimplePie's `get_items` call.
 		 * The default number of tracks is ten.
 		 *
-		 * @since 9.5.0
+		 * @since 10.4.0
 		 *
-		 * @param int    $number Number of tracks fetched. Default is 10.
-		 * @param object $rss    The SimplePie object built from core's `fetch_feed` call.
+		 * @param int $number Number of tracks fetched. Default is 10.
 		 */
-		return apply_filters( 'jetpack_podcast_helper_list_quantity', 10, $rss );
+		return (int) apply_filters( 'jetpack_podcast_helper_tracks_quantity', 10 );
 	}
 
 	/**
@@ -200,6 +193,19 @@ class Jetpack_Podcast_Helper {
 
 		$tracks_quantity = $this->get_track_quantity();
 
+		/**
+		 * Allow requesting a specific number of tracks from SimplePie's `get_items` call.
+		 * The default number of tracks is ten.
+		 * Deprecated. Use jetpack_podcast_tracks_quantity filter instead, which takes one less parameter.
+		 *
+		 * @since 9.5.0
+		 * @deprecated 10.4.0
+		 *
+		 * @param int    $tracks_quantity Number of tracks fetched. Default is 10.
+		 * @param object $rss             The SimplePie object built from core's `fetch_feed` call.
+		 */
+		$tracks_quantity = apply_filters_deprecated( 'jetpack_podcast_helper_list_quantity', array( $tracks_quantity, $rss ), '10.4.0', 'jetpack_podcast_helper_tracks_quantity' );
+
 		// Process the requested number of items from our feed.
 		$track_list = array_map( array( __CLASS__, 'setup_tracks_callback' ), $rss->get_items( 0, $tracks_quantity ) );
 
```

Reply:
Proposed changes make more sense to me and it will be easier to implement as well. We can start deprecating. 

One doubt is: Where is the filter definition `jetpack_podcast_helper_list_quantity`? I didn't find it in the codebase.

Reply:
I will try proposed changes in my branch and will see if any new issue comes or not. In my opinion deprecating above fitler with `$rss` should not impact at least this issue.

Reply:
> Where is the filter definition jetpack_podcast_helper_list_quantity?

From what I see the filter only exists in that `class-jetpack-podcast-helper.php` file. If `add_filter()` is never called, then `apply_filters( $hook, $value )` will return `$value` by default. So it allows for hooking into the filter without needing an explicit `add_filter()` declaration. 

Let us know if any questions come up or when this is ready for another look.

Reply:
@samiff, I have made the requested changes. thanks for clarifying `add_filter()`. It is working as expected.
Request you to please take an another round of look at the PR.